src/Collector.cs

```diff
@@ -93,17 +93,11 @@ private void WorkerOnDoWork(object sender, DoWorkEventArgs doWorkEventArgs)
           try
             {
               socket = Connect();
-              if(!Authenticate(socket))
-                {
-                  socket.Close();
-                  // Throttle calls to authenticate so that you do not flood the buffer if collector is down/slow for a few seconds.
-                  // Ultimately should share logic with the backoff stuff below, but did not want to just throw here.
-                  Thread.Sleep(5000);
-                  continue;
-                }
+              Authenticate(socket);
               SendQueuedMessages(socket);
-              failures = 0;
+              socket.Shutdown(SocketShutdown.Both);
               socket.Close();
+              failures = 0;
```

Comment:
If we move it to right after authenticate, and there's an issue with sending metrics but not with authenticate, the error rate will never back off though.